core/client/src/main/java/alluxio/client/block/UnderStoreBlockInStream.java

```diff
@@ -85,7 +93,7 @@ public long remaining() {
   public void seek(long pos) throws IOException {
     long offset = mPos - mInitPos;
     if (pos < offset) {
-      setUnderStoreStream(pos);
+      setUnderStoreStream(mInitPos + pos);
```

Comment:
I would prefer the block indexing position to be handled in setUnderStoreStream, and the guarantee to be the under store stream is always positioned correctly, ie. it shouldn't be possible to open the under store stream at a position which is not within the block.